File: Data Structures & Algorithms/subsets-ii/submission-0.cpp
class Solution {
   public:
    vector<vector<int>> subsetsWithDup(vector<int>& nums) {
        vector<vector<int>> res;
        sort(nums.begin(), nums.end());
        vector<int> subsets;
        dfs(nums, 0, subsets, res);
        return res;
    }

   private:
    void dfs(vector<int>& nums, int i, vector<int>& subsets, vector<vector<int>>& res) {
        if (i == nums.size()) {
            res.push_back(subsets);
            return;
        }
        subsets.push_back(nums[i]);
        dfs(nums, i + 1, subsets, res);
        subsets.pop_back();
        int idx = i + 1;
        while (idx < nums.size() && nums[idx] == nums[idx - 1]) {
            idx += 1;
        }
        dfs(nums, idx, subsets, res);
    }
};

File: Data Structures & Algorithms/find-first-and-last-position-of-element-in-sorted-array/submission-0.py
class Solution:
    def searchRange(self, nums: List[int], target: int) -> List[int]:
        l,r=0,len(nums)-1
        res=[-1,-1]
        while l<=r:
            m=l+(r-l)//2   
            if nums[m]>=target:
                r=m-1
            else:
                l=m+1
            if nums[m]==target:
                res[0]=m

        l,r=0,len(nums)-1
        while l<=r:
            m=l+(r-l)//2 
            if nums[m]<=target:
                l=m+1
            else:
                r=m-1
            if nums[m]==target:
                res[1]=m
        
        return res    